### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by dsapala**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -10,31 +10,25 @@ import (
 )
 
 var (
-	serverPort string
-	serverHost string
-	connType   string
+	serverHost = flag.String("host", "localhost", "server host")
```
Comment (mccurdyc) [inline main.go]: This is pretty sweet. I haven't seen this.

File: server/server.go [hunk 2]
```diff
@@ -13,11 +14,13 @@ type Service struct {
 	Server   http.Server
 }
 
-func NewService(p string) *Service {
+func NewService(host string, port int) *Service {
+	addr := fmt.Sprintf("%s:%d", host, port)
+
```
Comment (mccurdyc) [inline server/server.go]: This is exactly what I thought you would do.

**Comment by dsapala** [comment]
I put the vendor code into the repo so I don’t have to worry about the third party code not being there when I want it. It also makes sure that when I put this through a deployment pipeline I speed up the process and don’t have to worry about a dependency being unavailable.

**Comment by dsapala** [comment]
Having it in there also ensures that someone else who clones this repo can just run it

**Pull request merged**